File: macosx/ExpandedPathToPathTransformer.m

#import "ExpandedPathToPathTransformer.h"

@implementation ExpandedPathToPathTransformer

+ (Class) transformedValueClass
{
    return [NSString class];
}

+ (BOOL) allowsReverseTransformation
{
    return NO;
}

- (id) transformedValue: (id) value
{
    return value == nil ?  nil : [value lastPathComponent];
}

@end

File: macosx/ExpandedPathToPathTransformer.swift
// This file Copyright © 2007-2023 Transmission authors and contributors.
// It may be used under the MIT (SPDX: MIT) license.
// License text can be found in the licenses/ folder.

import Foundation

class ExpandedPathToPathTransformer: ValueTransformer {
    override class func transformedValueClass() -> AnyClass {
        return NSString.self
    }

    override class func allowsReverseTransformation() -> Bool {
        return false
    }

    override func transformedValue(_ value: Any?) -> Any? {
        guard let value = value as? String else {
            return nil
        }
        return FileManager.default.displayName(atPath: value)
    }
}
